daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -286,23 +413,28 @@ class VariableMap private(vTable: Map[GlobalQName, MStackOf[VariableInstance]])
   /**
    * Creates a new instance of a variable
    */
-  def newVariableInstance(vrd: VariableRuntimeData) = {
+  def newVariableInstance(vrd: VariableRuntimeData, referringContext: ThrowsSDE, state: ParseOrUnparseState) = {
     val varQName = vrd.globalQName
-    val stack = vTable.get(varQName)
-    Assert.invariant(stack.isDefined)
-    stack.get.push(vrd.createVariableInstance)
+    val abuf = vTable.get(varQName)
+    Assert.invariant(abuf.isDefined)
+
+    if (vrd.maybeDefaultValueExpr.isDefined) {
+      val defaultValue = DataValue.unsafeFromAnyRef(vrd.maybeDefaultValueExpr.get.evaluate(state))
+      abuf.get += vrd.createVariableInstance(VariableUtils.convert(defaultValue.getAnyRef.toString, vrd, referringContext))
```

Comment:
This ``VariableUtils.convert`` seems incorrect to me. I would expect that evaluating the defaultValue expression, or any setVariable expression as well, would know the target type and the expression evaluation would handle the conversion, just like it does with all other expression evaluations. Looks like this isn't something you added, so probably out of scope for this change. But we should create a bug to fix this.

Reply:
I've pushed up changes addressing all your other comments and I've created DAFFODIL-2445 to track this issue.